src/Gui/MenuManager.cpp

```diff
@@ -299,77 +298,73 @@ void MenuManager::setup(MenuItem* menuItems) const
+    for (auto* item : item->getItems()) {
         // search for the menu item
-        QList<QAction*> used_actions = findActions(actions, QString::fromLatin1((*it)->command().c_str()));
+        QList<QAction*> used_actions = findActions(actions, QString::fromLatin1(item->command().c_str()));
         if (used_actions.isEmpty()) {
-            if ((*it)->command() == "Separator") {
+            if (item->command() == "Separator") {
                 QAction* action = menu->addSeparator();
                 action->setObjectName(QLatin1String("Separator"));
-                // set the menu user data
                 action->setData(QLatin1String("Separator"));
                 used_actions.append(action);
             }
             else {
-                if ((*it)->hasItems()) {
+                if (item->hasItems()) {
                     // Creste a submenu
-                    std::string menuName = (*it)->command();
-                    QMenu* submenu = menu->addMenu(
-                        QApplication::translate("Workbench", menuName.c_str()));
+                    std::string menuName = item->command();
+                    QMenu* submenu = menu->addMenu(QApplication::translate("Workbench", menuName.c_str()));
                     QAction* action = submenu->menuAction();
-                    submenu->setObjectName(QString::fromLatin1((*it)->command().c_str()));
-                    action->setObjectName(QString::fromLatin1((*it)->command().c_str()));
+                    submenu->setObjectName(QString::fromLatin1(item->command().c_str()));
+                    action->setObjectName(QString::fromLatin1(item->command().c_str()));
                     // set the menu user data
-                    action->setData(QString::fromLatin1((*it)->command().c_str()));
+                    action->setData(QString::fromLatin1(item->command().c_str()));
                     used_actions.append(action);
                 }
                 else {
                     // A command can have more than one QAction
                     int count = menu->actions().count();
                     // Check if action was added successfully
-                    if (mgr.addTo((*it)->command().c_str(), menu)) {
+                    if (mgr.addTo(item->command().c_str(), menu)) {
                         QList<QAction*> acts = menu->actions();
                         for (int i=count; i < acts.count(); i++) {
                             QAction* act = acts[i];
                             // set the menu user data
-                            act->setData(QString::fromLatin1((*it)->command().c_str()));
+                            act->setData(QString::fromLatin1(item->command().c_str()));
                             used_actions.append(act);
                         }
                     }
                 }
             }
         }
         else {
-            for (QList<QAction*>::Iterator it = used_actions.begin(); it != used_actions.end(); ++it) {

```

Comment:
Note: Here the same variable name 'it' was used in this for loop but also in the parent for loop. Which was maybe or maybe not generating bugs. Or it could also be that it was preventing a bug from happening. If you could take a quick look at it @wwmayer please ?

Reply:
To be on the safe side you can use a new iterator variable. But it's probably better to refactor this function to avoid all the cascaded if-else by moving the code to some helper functions or lambdas.